[explore-v2] make chart container work with existing visualization files

ascott opened the pull request:
- makes `<ChartContainer />` use existing visualization js files to render charts.

todo:
- [x] use vizMap object to map chart type to vis module
  https://github.com/ascott/caravel/blob/master/caravel/assets/visualizations/main.js#L3
- [x] figure out if other vis modules have other different dependencies than than the `nvd3_vis` module
  ---- added an issue documenting all the issues with wrapping vis's in V2. will come back for these in a follow up PR. https://github.com/airbnb/caravel/issues/1341

/exploreV2/ with line, bar, area, pie, bubble, box_plot

![screenshot 2016-10-12 18 15 36](https://cloud.githubusercontent.com/assets/130878/19333228/e0a250a6-90a8-11e6-8c73-09314b98871c.png)
![screenshot 2016-10-12 18 15 57](https://cloud.githubusercontent.com/assets/130878/19333232/e0a4f2fc-90a8-11e6-84e0-e16aafcd7dac.png)
![screenshot 2016-10-12 18 16 21](https://cloud.githubusercontent.com/assets/130878/19333230/e0a40f4a-90a8-11e6-9eb7-f658103e4a7a.png)
![screenshot 2016-10-12 18 17 06](https://cloud.githubusercontent.com/assets/130878/19333229/e0a393da-90a8-11e6-9fd9-02c12c7d7729.png)
![screenshot 2016-10-12 18 17 34](https://cloud.githubusercontent.com/assets/130878/19333233/e0a6c92e-90a8-11e6-90de-86f3da9b0d91.png)
![screenshot 2016-10-12 18 18 08](https://cloud.githubusercontent.com/assets/130878/19333231/e0a4b86e-90a8-11e6-94c9-c0b7beaac81a.png)

cc @mistercrunch @vera-liu

Commit:
make chart container work with nvd3_vis.js

mistercrunch commented on caravel/assets/javascripts/explorev2/components/ChartContainer.jsx:
```diff
@@ -3,28 +3,63 @@ import { connect } from 'react-redux';
 import { Panel } from 'react-bootstrap';
 import TimeSeriesLineChart from './charts/TimeSeriesLineChart';
 import moment from 'moment';
+import nvd3Vis from '../../../visualizations/nvd3_vis';
```
To generalize here you'll be able to import the object that maps the visualization type string to the object, it's in `../../../visualizations/nvd3_vis`

mistercrunch commented on caravel/assets/javascripts/explorev2/components/ChartContainer.jsx:
```diff
@@ -3,28 +3,63 @@ import { connect } from 'react-redux';
-    return this.props.vizType === 'line';
+  renderVis() {
+    const slice = this.getMockedSliceObject();
+    nvd3Vis(slice).render();
```
then here use: `vizMap[vizType](slice).render()`

Commit:
map vis to module, remove unneeded components

Commit:
fix linting

Commit:
use existing query and save btns, don't fork more things

ascott commented:
table type has a dependency on slice.d3format:

<img width="1280" alt="screenshot 2016-10-13 14 36 06" src="https://cloud.githubusercontent.com/assets/130878/19367958/7e268bea-9152-11e6-9c05-e41bf85121fc.png">

ascott commented:
big number with trendline doesn't show big number

![screenshot 2016-10-13 14 41 53](https://cloud.githubusercontent.com/assets/130878/19368099/437f2c58-9153-11e6-8eee-3cedea901a06.png)

ascott commented:
big number doesn't render big number:

![screenshot 2016-10-13 14 42 55](https://cloud.githubusercontent.com/assets/130878/19368134/6ccb781e-9153-11e6-95df-113f42f63068.png)
![screenshot 2016-10-13 14 43 07](https://cloud.githubusercontent.com/assets/130878/19368133/6cc44fda-9153-11e6-9fa6-7f4e2e19e619.png)

ascott commented:
directed force layout looks a little different.... 

![screenshot 2016-10-13 14 44 39](https://cloud.githubusercontent.com/assets/130878/19368178/a94075a6-9153-11e6-9b4e-57ee74ed8c7a.png)
![screenshot 2016-10-13 14 44 54](https://cloud.githubusercontent.com/assets/130878/19368179/a94b6c0e-9153-11e6-84ee-13e4d03ac33c.png)

ascott commented:
region filter is broken on /explore/

![screenshot 2016-10-13 14 46 20](https://cloud.githubusercontent.com/assets/130878/19368207/d7809c84-9153-11e6-8a5d-8e79f9252440.png)

/exploerV2/ has dependency on `slice.getFilters`
![screenshot 2016-10-13 14 47 07](https://cloud.githubusercontent.com/assets/130878/19368236/fd2bb36a-9153-11e6-893b-27e4fcff02f1.png)

ascott commented:
heat map has some layout issues:
![screenshot 2016-10-13 14 48 12](https://cloud.githubusercontent.com/assets/130878/19368284/27d0cd3a-9154-11e6-9ab0-f546dd37ff88.png)
![screenshot 2016-10-13 14 48 28](https://cloud.githubusercontent.com/assets/130878/19368285/27d13bd0-9154-11e6-8355-b1d37af83ad7.png)

ascott commented:
pivot table type throws an error in V2:

![screenshot 2016-10-13 14 52 57](https://cloud.githubusercontent.com/assets/130878/19368444/dc1cbcf4-9154-11e6-992b-7012e7075895.png)
![screenshot 2016-10-13 14 53 24](https://cloud.githubusercontent.com/assets/130878/19368445/dc1cfc46-9154-11e6-8d36-77d3392a6024.png)

ascott commented:
sankey as layout/visual issues in V2:

![screenshot 2016-10-13 14 54 17](https://cloud.githubusercontent.com/assets/130878/19368476/03186754-9155-11e6-9131-247a05f0611a.png)
![screenshot 2016-10-13 14 54 36](https://cloud.githubusercontent.com/assets/130878/19368477/0318c4b0-9155-11e6-8c1b-649350c645c8.png)

ascott commented:
sunburst has layout/visual issues in V2:
![screenshot 2016-10-13 14 55 16](https://cloud.githubusercontent.com/assets/130878/19368525/39994c12-9155-11e6-98a4-22d096b89632.png)
![screenshot 2016-10-13 14 55 59](https://cloud.githubusercontent.com/assets/130878/19368526/3999c1e2-9155-11e6-8d7b-29986dab8bb3.png)

ascott commented:
word_map type throws an error in V2:

![screenshot 2016-10-13 14 58 35](https://cloud.githubusercontent.com/assets/130878/19368601/99a18688-9155-11e6-9431-0e7a161c33d5.png)
![screenshot 2016-10-13 14 58 49](https://cloud.githubusercontent.com/assets/130878/19368600/99a0d332-9155-11e6-9930-d5c1e3008e63.png)

Commit:
comment out chart and exploreviecontainer specs

ascott commented:
@mistercrunch added support to use the mapping from vis type to vis file. i removed some components I had duplicated in explorev2, and also removed the new line chart in favor of using the existing viz files. i added an issue documenting the outstanding issues with wrapping the vis's in V2 and will make another PR with fixes for those. https://github.com/airbnb/caravel/issues/1341. PTAL!

Commit:
make a change because i think the js test is failing spuriously
